fetch/fetch.go

```diff
@@ -271,6 +271,7 @@ func NewFetch(
 ) *Fetch {
 	bs := datastore.NewBlobStore(cdb, proposals)
 
+	hashPeerCache, _ := NewHashPeersCache(cacheSize) // only errors if `cacheSize` is zero or negative
```

Comment:
I'd rather it panicked instead of silently swallowing the error.  E.g. somebody might want to switch caching for debugging or whatever and set `cacheSize = 0`, what would happen then?

Reply:
Fair point, I updated the code to also return the error in this case